### ISSUE TITLE
Incorrect Thresholds and Confidence Bands for Regression Metrics

### ISSUE THREAD

**Issue by nikml** [NannyML/nannyml#127]
**Describe the bug**
The thresholds and confidence bands for some regression metrics can 

![newplot](https://user-images.githubusercontent.com/89025229/191553988-c06a754f-8014-4841-b9d5-ba7307978aee.png)

What is wrong with the above plot? 
- The lower threshold for performance change is below 0. That does not make sense all regression performance metrics as they can't take negative values.
- The lower confidence band on the last chunk goes below 0. Again that doesn't make sense. Sampling error cannot make a performance metric take unphysical values (ie values that it can't take).

**To Reproduce**
1. Download the UCI Superconductivity data available [here](https://archive.ics.uci.edu/ml/datasets/Superconductivty+Data)
2. Run the following code:

```
import datetime as dt
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import json
import boto3
import nannyml as nml
from sklearn.ensemble import GradientBoostingRegressor
from sklearn.metrics import mean_absolute_error
import plotly.graph_objects as go

# change location below as appropriate for your machine
data = pd.read_csv("/var/home/nannyml/Downloads/superconduct/train.csv", header=0)

features = list(data.columns)[:-1]
data = data.assign(partition = 'train')
data.loc[data.shape[0]//3:, 'partition'] = 'reference'
data.loc[data.shape[0]//3+1:(data.shape[0] - data.shape[0]//3), 'partition'] = 'analysis'


gbm = GradientBoostingRegressor(random_state=14)
gbm.fit(
    X=data.loc[data.partition == 'train', features],
    y=data.loc[data.partition == 'train', 'critical_temp']
)
data = data.assign(y_pred = gbm.predict(X=data[features]))


reference = data.loc[data.partition == 'reference', :].reset_index(drop=True)
analysis = data.loc[data.partition == 'analysis', :].reset_index(drop=True)

estimator = nml.DLE(
    feature_column_names=features,
    y_pred='y_pred',
    y_true='critical_temp',
    # timestamp_column_name='timestamp',
    metrics=['mae', 'mse'],
    chunk_size=data.shape[0]//30,
    tune_hyperparameters=False
)

estimator.fit(reference)
results = estimator.estimate(analysis)

metric_fig = results.plot(kind='performance', metric='mse', plot_reference=False)
metric_fig.show()
```

3. Inspect Resulting plot

**Expected behavior**
- There would be no lower threshold - given that a threshold of 0 doesn't make sense.
- The sampling error number on the hover would stay the same. But the shaded area of the confidence band would only go up to 0, not below.

**Comment by kshitiz305** [comment]
HI @all
I am a python developer for the last four years and am looking forward to contribute to some of the open source. I have experience in building some data oriented products using python as the main base programming language. Additionally I have hand on experience in FastAPI. I am a quick learner and also have some bandwidth to contribute to the project.

May I know from where I setup the code and start my contribution process.
